contracts/light-clients/AttestorLightClient.sol

```diff
@@ -0,0 +1,216 @@
+        external
+        view
+        notFrozen
+        onlyProofSubmitter
+        returns (uint256)
+    {
+        if (msg_.value.length == 0) revert EmptyValue();
+
+        // Ensure we have a trusted timestamp at the provided height.
+        uint64 height = msg_.proofHeight.revisionHeight;
+        uint64 ts = _consensusTimestampAtHeight[height];
+        if (ts == 0) revert ConsensusTimestampNotFound(height);
+
+        IAttestorMsgs.AttestationProof memory proof = abi.decode(msg_.proof, (IAttestorMsgs.AttestationProof));
+        bytes32 digest = sha256(proof.attestationData);
+        _verifySignatures(digest, proof.signatures);
+
+        // Decode the attested packet commitments and verify the attested height matches the provided proof height
+        IAttestorMsgs.PacketAttestation memory packetAttestation =
+            abi.decode(proof.attestationData, (IAttestorMsgs.PacketAttestation));
+
+        if (packetAttestation.height != height) revert HeightMismatch(height, packetAttestation.height);
+
+        // Check membership: value must be present in the attested list
+        bool found = false;
+        bytes32 value = abi.decode(msg_.value, (bytes32));
+        for (uint256 i = 0; i < packetAttestation.packets.length; ++i) {
+            if (packetAttestation.packets[i] == value) {
+                found = true;
+                break;
+            }
+        }
+        if (!found) revert NotMember();
+
+        return uint256(ts);
+    }
+
+    /// @inheritdoc ILightClient
+    function verifyNonMembership(ILightClientMsgs.MsgVerifyNonMembership calldata)
+        external
+        view
+        notFrozen
+        onlyProofSubmitter
+        returns (uint256)
+    {
+        // Out of scope for this version
+        revert FeatureNotSupported();
+    }
+
+    /// @inheritdoc ILightClient
+    function misbehaviour(bytes calldata) external view notFrozen onlyProofSubmitter {
+        // Out of scope for this version
+        revert FeatureNotSupported();
+    }
+
+    /// @inheritdoc ILightClient
+    function upgradeClient(bytes calldata) external view notFrozen onlyProofSubmitter {
+        revert FeatureNotSupported();
+    }
+
+    /// @notice Verifies that `signatures` over `digest` are valid, unique, and meet the threshold.
+    /// @param digest The message hash that attestors must have signed.
+    /// @param signatures Compact ECDSA signatures (r||s||v) provided by attestors.
+    /// @dev Reverts with `InvalidSignatureLength`, `SignatureInvalid`, `UnknownSigner`, `DuplicateSigner`,
+    ///      or `ThresholdNotMet` on failure.
+    function _verifySignatures(bytes32 digest, bytes[] memory signatures) private view {
+        address[] memory seen = new address[](signatures.length);
+        uint256 seenLen = 0;
+
+        uint256 valid = 0;
+        for (uint256 i = 0; i < signatures.length; ++i) {
+            bytes memory sig = signatures[i];
+            if (sig.length != 65) revert InvalidSignatureLength(i);
```

Comment:
This is actually checked in `ECDSA.recover`, however, this check might be removed in openzeppelin v6. So, it might be worth keeping it here